drivers/i2c/i2c_omap.c

```diff
@@ -0,0 +1,820 @@
+#define I2C_OMAP_CON_STT       BIT(0)  /* Start condition (controller) */
+
+/* I2C_OMAP_WE wakeup enable register */
+#define I2C_OMAP_WE_XDR_WE  BIT(14) /* TX drain wakeup */
+#define I2C_OMAP_WE_RDR_WE  BIT(13) /* RX drain wakeup */
+#define I2C_OMAP_WE_AAS_WE  BIT(9)  /* Address as target wakeup */
+#define I2C_OMAP_WE_BF_WE   BIT(8)  /* Bus free wakeup */
+#define I2C_OMAP_WE_STC_WE  BIT(6)  /* Start condition wakeup */
+#define I2C_OMAP_WE_GC_WE   BIT(5)  /* General call wakeup */
+#define I2C_OMAP_WE_DRDY_WE BIT(3)  /* TX/RX data ready wakeup */
+#define I2C_OMAP_WE_ARDY_WE BIT(2)  /* Register access ready wakeup */
+#define I2C_OMAP_WE_NACK_WE BIT(1)  /* No acknowledgment wakeup */
+#define I2C_OMAP_WE_AL_WE   BIT(0)  /* Arbitration lost wakeup */
+
+#define I2C_OMAP_WE_ALL                                                                            \
+	(I2C_OMAP_WE_XDR_WE | I2C_OMAP_WE_RDR_WE | I2C_OMAP_WE_AAS_WE | I2C_OMAP_WE_BF_WE |        \
+	 I2C_OMAP_WE_STC_WE | I2C_OMAP_WE_GC_WE | I2C_OMAP_WE_DRDY_WE | I2C_OMAP_WE_ARDY_WE |      \
+	 I2C_OMAP_WE_NACK_WE | I2C_OMAP_WE_AL_WE)
+
+/* I2C Buffer Configuration Register (I2C_OMAP_BUF): */
+#define I2C_OMAP_BUF_RXFIF_CLR BIT(14) /* RX FIFO Clear */
+#define I2C_OMAP_BUF_TXFIF_CLR BIT(6)  /* TX FIFO Clear */
+
+/* I2C Status Register (I2C_OMAP_STAT): */
+#define I2C_OMAP_STAT_XDR  BIT(14) /* TX Buffer draining */
+#define I2C_OMAP_STAT_RDR  BIT(13) /* RX Buffer draining */
+#define I2C_OMAP_STAT_BB   BIT(12) /* Bus busy */
+#define I2C_OMAP_STAT_ROVR BIT(11) /* Receive overrun */
+#define I2C_OMAP_STAT_XUDF BIT(10) /* Transmit underflow */
+#define I2C_OMAP_STAT_AAS  BIT(9)  /* Address as target */
+#define I2C_OMAP_STAT_XRDY BIT(4)  /* Transmit data ready */
+#define I2C_OMAP_STAT_RRDY BIT(3)  /* Receive data ready */
+#define I2C_OMAP_STAT_ARDY BIT(2)  /* Register access ready */
+#define I2C_OMAP_STAT_NACK BIT(1)  /* No ack interrupt enable */
+#define I2C_OMAP_STAT_AL   BIT(0)  /* Arbitration lost */
+
+/* I2C System Test Register (I2C_OMAP_SYSTEST): */
+#define I2C_OMAP_SYSTEST_ST_EN       BIT(15)   /* System test enable */
+#define I2C_OMAP_SYSTEST_FREE        BIT(14)   /* Free running mode */
+#define I2C_OMAP_SYSTEST_TMODE_MASK  (3 << 12) /* Test mode select mask */
+#define I2C_OMAP_SYSTEST_TMODE_SHIFT (12)      /* Test mode select shift */
+
+/* Functional mode */
+#define I2C_OMAP_SYSTEST_SCL_I_FUNC BIT(8) /* SCL line input value */
+#define I2C_OMAP_SYSTEST_SDA_I_FUNC BIT(6) /* SDA line input value */
+
+/* SDA/SCL IO mode */
+#define I2C_OMAP_SYSTEST_SCL_I BIT(3) /* SCL line sense in */
+#define I2C_OMAP_SYSTEST_SCL_O BIT(2) /* SCL line drive out */
+#define I2C_OMAP_SYSTEST_SDA_I BIT(1) /* SDA line sense in */
+#define I2C_OMAP_SYSTEST_SDA_O BIT(0) /* SDA line drive out */
+
+typedef void (*init_func_t)(const struct device *dev);
+#define DEV_I2C_BASE(dev) ((I2C_OMAP_Type *)DEVICE_MMIO_NAMED_GET(dev, base))
+
```

Comment:
You could consider to add more a define for data and config. See https://github.com/zephyrproject-rtos/zephyr/blob/31fea97e05fd524b089c70281356d07df7788349/drivers/gpio/gpio_davinci.c#L28

It's not a huge improvement but in case a struct changes, this has to be changed only at one location.

Reply:
This makes sense.
Thanks for showing this.
Will update it!